Rewrite for Mopidy 3

djmattyg007 opened the pull request:
This PR is almost a complete rewrite. It updates the code for Mopidy v3, as well as updating the frontend code for modern standards.

The Python side of things has been updated to conform to modern Mopidy extension standards. Most things in `setup.py` have been moved into a `setup.cfg` file, for example. Most extensions put their static web files into a directory named `static`, so this was also adjusted. The extension init code uses pathlib, and a redundant method override has been cleaned up.

On the frontend side, Angular has been completely removed. Handlebars is a much more lightweight solution for templating, so this is now used. Bootstrap has been upgraded to v5, which allows us to remove jQuery. The Angular Bootstrap integration is obviously also no longer necessary. Also, Mopidy.js is no longer bundled with Mopidy, so that has been vendored.

djmattyg007 committed:
Initial re-work for Mopidy 3

djmattyg007 committed:
Massive rewrite and simplification

- Remove Angular
- Update bootstrap to latest version, remove jQuery
- Use Handlebars for templating needs
- Improve scrollspy integration

djmattyg007 committed:
Minor CSS improvement

djmattyg007 committed:
Minor presentation improvement

djmattyg007 committed:
Update readme

kingosticks commented:
This looks really good to me. The style is a bit different but that will just take time to get used to.

I did notice that if you skip to a section/method it scrolls you just a little too far and the heading for the link you just clicked is obscured by the navbar. Without being able to see the heading I wasn't totally sure it had taken taken me to the right place. You can of course scroll up one, or just trust it's done the right thing and then it's not a problem. This is with Firefox 84.0.1 (64-bit) for Ubuntu.

djmattyg007 commented:
Thanks for the feedback, I appreciate it!

I had noticed the scrolling issue, but it was literally 2am when I submitted this PR and I needed sleep :P I'll see what I can do about it at some point today.

djmattyg007 committed:
Code style cleanup

djmattyg007 committed:
Resolve scrolling issues

This makes use of a relatively new part of CSS known as "Scroll Snap".
It's well-supported in all modern browsers, which should be fine for a
developer tool.

I'm not completely sure how much the Bootstrap Scrollspy offset change
is needed, but it seems to work well in its current state.

djmattyg007 commented:
@kingosticks I've just pushed an update and would appreciate it if you could give it a try. It should hopefully resolve the scrolling issue you highlighted.

djmattyg007 commented:
Thanks for encouraging me to take a look at this btw, I didn't expect to learn something new about CSS and was expecting to have to fiddle with some weird JS code.

kingosticks commented:
That now works perfectly, awesome!

djmattyg007 committed:
Further readme updates

djmattyg007 committed:
Another minor readme update
